packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx

```diff
@@ -171,12 +173,18 @@ export function DayPicker<TDate>(props: DayPickerProps<TDate>) {
   const transitionKey = currentMonthNumber;
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const slideNodeRef = React.useMemo(() => React.createRef<HTMLDivElement>(), [transitionKey]);
+  const startOfCurrentWeek = utils.startOfWeek(now);
 
   return (
-    <React.Fragment>
-      <PickersCalendarDayHeader>
+    <div role="grid" aria-labelledby={gridLabelId}>
```

Comment:
A small detail but as MUI convention we should spread all the remaining props to the root element.

Reply:
I can add this logic in, but at least TS wise, there are no unused properties left in this component. 🤔

Reply:
Your suggestion would require quite a few other changes to avoid unwanted props falling through to the `DayPicker` root div element. 🤔 
I'm not sure if we should go with this change on this PR, what do you think?

```diff
diff --git a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerView.tsx b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerView.tsx
index 5075360a..fbde7bb8 100644
--- a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerView.tsx
+++ b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerView.tsx
@@ -126,24 +126,19 @@ function DateRangePickerViewRaw<TInputDate, TDate>(
   const wrapperVariant = React.useContext(WrapperVariantContext);
 
   const [start, end] = parsedValue;
-  const {
-    changeMonth,
-    calendarState,
-    isDateDisabled,
-    onMonthSwitchingAnimationEnd,
-    changeFocusedDay,
-  } = useCalendarState({
-    date: start || end,
-    defaultCalendarMonth,
-    disableFuture,
-    disablePast,
-    disableSwitchToMonthOnDayFocus: true,
-    maxDate,
-    minDate,
-    onMonthChange,
-    reduceAnimations,
-    shouldDisableDate,
-  });
+  const { changeMonth, calendarState, isDateDisabled, onMonthSwitchingAnimationEnd } =
+    useCalendarState({
+      date: start || end,
+      defaultCalendarMonth,
+      disableFuture,
+      disablePast,
+      disableSwitchToMonthOnDayFocus: true,
+      maxDate,
+      minDate,
+      onMonthChange,
+      reduceAnimations,
+      shouldDisableDate,
+    });
 
   const toShowToolbar = showToolbar ?? wrapperVariant !== 'desktop';
 
@@ -218,13 +213,11 @@ function DateRangePickerViewRaw<TInputDate, TDate>(
   const renderView = () => {
     const sharedCalendarProps = {
       parsedValue,
-      changeFocusedDay,
       onSelectedDaysChange: handleSelectedDayChange,
       reduceAnimations,
       disableHighlightToday,
       onMonthSwitchingAnimationEnd,
       changeMonth,
-      currentlySelectingRangeEnd,
       disableFuture,
       disablePast,
       minDate,
@@ -236,7 +229,13 @@ function DateRangePickerViewRaw<TInputDate, TDate>(
 
     switch (wrapperVariant) {
       case 'desktop': {
-        return <DateRangePickerViewDesktop calendars={calendars} {...sharedCalendarProps} />;
+        return (
+          <DateRangePickerViewDesktop
+            calendars={calendars}
+            currentlySelectingRangeEnd={currentlySelectingRangeEnd}
+            {...sharedCalendarProps}
+          />
+        );
       }
 
       default: {
diff --git a/packages/x-date-pickers-pro/src/DesktopDateRangePicker/DesktopDateRangePicker.tsx b/packages/x-date-pickers-pro/src/DesktopDateRangePicker/DesktopDateRangePicker.tsx
index 957345d8..a7d36683 100644
--- a/packages/x-date-pickers-pro/src/DesktopDateRangePicker/DesktopDateRangePicker.tsx
+++ b/packages/x-date-pickers-pro/src/DesktopDateRangePicker/DesktopDateRangePicker.tsx
@@ -96,11 +96,24 @@ export const DesktopDateRangePicker = React.forwardRef(function DesktopDateRange
     TransitionComponent,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
+    DialogProps,
+    onAccept,
+    onOpen,
+    closeOnSelect,
     ...other
   } = props;
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
+    onAccept,
+    onOpen,
+    closeOnSelect,
     components,
     componentsProps,
     currentlySelectingRangeEnd,
diff --git a/packages/x-date-pickers-pro/src/MobileDateRangePicker/MobileDateRangePicker.tsx b/packages/x-date-pickers-pro/src/MobileDateRangePicker/MobileDateRangePicker.tsx
index 4469c901..a614eb5a 100644
--- a/packages/x-date-pickers-pro/src/MobileDateRangePicker/MobileDateRangePicker.tsx
+++ b/packages/x-date-pickers-pro/src/MobileDateRangePicker/MobileDateRangePicker.tsx
@@ -78,7 +78,20 @@ export const MobileDateRangePicker = React.forwardRef(function MobileDateRangePi
     MobileDateRangePickerProps<TInputDate, TDate>
   >(inProps, 'MuiMobileDateRangePicker');
 
-  const { value, onChange, components, componentsProps, ...other } = props;
+  const {
+    value,
+    onChange,
+    components,
+    componentsProps,
+    label,
+    inputFormat,
+    renderInput,
+    DialogProps,
+    onOpen,
+    onAccept,
+    closeOnSelect,
+    ...other
+  } = props;
 
   const [currentlySelectingRangeEnd, setCurrentlySelectingRangeEnd] = React.useState<
     'start' | 'end'
@@ -100,6 +113,12 @@ export const MobileDateRangePicker = React.forwardRef(function MobileDateRangePi
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
+    onOpen,
+    onAccept,
+    closeOnSelect,
     components,
     componentsProps,
     currentlySelectingRangeEnd,
@@ -116,6 +135,7 @@ export const MobileDateRangePicker = React.forwardRef(function MobileDateRangePi
       PureDateInputComponent={PureDateInputComponent}
       components={components}
       componentsProps={componentsProps}
+      DialogProps={DialogProps}
     >
       <DateRangePickerView
         open={wrapperProps.open}
diff --git a/packages/x-date-pickers-pro/src/StaticDateRangePicker/StaticDateRangePicker.tsx b/packages/x-date-pickers-pro/src/StaticDateRangePicker/StaticDateRangePicker.tsx
index d4e8507d..7c54e5a1 100644
--- a/packages/x-date-pickers-pro/src/StaticDateRangePicker/StaticDateRangePicker.tsx
+++ b/packages/x-date-pickers-pro/src/StaticDateRangePicker/StaticDateRangePicker.tsx
@@ -91,12 +91,18 @@ export const StaticDateRangePicker = React.forwardRef(function StaticDateRangePi
     onChange,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
 
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     currentlySelectingRangeEnd,
     setCurrentlySelectingRangeEnd,
     validationError,
diff --git a/packages/x-date-pickers/src/CalendarPicker/CalendarPicker.tsx b/packages/x-date-pickers/src/CalendarPicker/CalendarPicker.tsx
index a1176953..22642b85 100644
--- a/packages/x-date-pickers/src/CalendarPicker/CalendarPicker.tsx
+++ b/packages/x-date-pickers/src/CalendarPicker/CalendarPicker.tsx
@@ -211,6 +211,7 @@ const CalendarPicker = React.forwardRef(function CalendarPicker<TDate>(
     minDate = defaultDates.minDate,
     maxDate = defaultDates.maxDate,
     disableHighlightToday,
+    componentsProps,
     ...other
   } = props;
 
@@ -401,6 +402,7 @@ const CalendarPicker = React.forwardRef(function CalendarPicker<TDate>(
         disableFuture={disableFuture}
         reduceAnimations={reduceAnimations}
         labelId={gridLabelId}
+        componentsProps={componentsProps}
       />
       <CalendarPickerViewTransitionContainer
         reduceAnimations={reduceAnimations}
diff --git a/packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx b/packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx
index 1008ca3d..06ff2dc6 100644
--- a/packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx
+++ b/packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx
@@ -143,6 +143,7 @@ export function DayPicker<TDate>(props: DayPickerProps<TDate>) {
     shouldDisableDate,
     dayOfWeekFormatter = defaultDayOfWeekFormatter,
     gridLabelId,
+    ...other
   } = props;
 
   const isDateDisabled = useIsDayDisabled({
@@ -176,7 +177,7 @@ export function DayPicker<TDate>(props: DayPickerProps<TDate>) {
   const startOfCurrentWeek = utils.startOfWeek(now);
 
   return (
-    <div role="grid" aria-labelledby={gridLabelId}>
+    <div role="grid" aria-labelledby={gridLabelId} {...other}>
       <PickersCalendarDayHeader role="row">
         {utils.getWeekdays().map((day, i) => (
           <PickersCalendarWeekDayLabel
diff --git a/packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePicker.tsx b/packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePicker.tsx
index 8921901b..b343e872 100644
--- a/packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePicker.tsx
+++ b/packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePicker.tsx
@@ -80,11 +80,17 @@ export const DesktopDatePicker = React.forwardRef(function DesktopDatePicker<
     value,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
   const AllDateInputProps = {
     ...inputProps,
     ...other,
+    inputFormat,
+    label,
+    renderInput,
     components,
     componentsProps,
     ref,
diff --git a/packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx b/packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx
index 8f76bad8..5c6f8ed4 100644
--- a/packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx
+++ b/packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx
@@ -86,11 +86,17 @@ export const DesktopDateTimePicker = React.forwardRef(function DesktopDateTimePi
     value,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
   const AllDateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     components,
     componentsProps,
     ref,
diff --git a/packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx b/packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx
index adc7eeeb..0b58b442 100644
--- a/packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx
+++ b/packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx
@@ -82,11 +82,17 @@ export const DesktopTimePicker = React.forwardRef(function DesktopTimePicker<
     value,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     components,
     componentsProps,
     ref,
diff --git a/packages/x-date-pickers/src/MobileDatePicker/MobileDatePicker.tsx b/packages/x-date-pickers/src/MobileDatePicker/MobileDatePicker.tsx
index 18ec172f..aa9101ed 100644
--- a/packages/x-date-pickers/src/MobileDatePicker/MobileDatePicker.tsx
+++ b/packages/x-date-pickers/src/MobileDatePicker/MobileDatePicker.tsx
@@ -79,11 +79,17 @@ export const MobileDatePicker = React.forwardRef(function MobileDatePicker<
     onChange,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     components,
     componentsProps,
     ref,
diff --git a/packages/x-date-pickers/src/MobileDateTimePicker/MobileDateTimePicker.tsx b/packages/x-date-pickers/src/MobileDateTimePicker/MobileDateTimePicker.tsx
index 8d98200c..99fa4964 100644
--- a/packages/x-date-pickers/src/MobileDateTimePicker/MobileDateTimePicker.tsx
+++ b/packages/x-date-pickers/src/MobileDateTimePicker/MobileDateTimePicker.tsx
@@ -83,11 +83,17 @@ export const MobileDateTimePicker = React.forwardRef(function MobileDateTimePick
     onChange,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     components,
     componentsProps,
     ref,
diff --git a/packages/x-date-pickers/src/MobileTimePicker/MobileTimePicker.tsx b/packages/x-date-pickers/src/MobileTimePicker/MobileTimePicker.tsx
index 5f2d69dc..635ae614 100644
--- a/packages/x-date-pickers/src/MobileTimePicker/MobileTimePicker.tsx
+++ b/packages/x-date-pickers/src/MobileTimePicker/MobileTimePicker.tsx
@@ -80,11 +80,17 @@ export const MobileTimePicker = React.forwardRef(function MobileTimePicker<
     onChange,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     components,
     componentsProps,
     ref,
diff --git a/packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.tsx b/packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.tsx
index 3a0a18c5..4ae4bcfb 100644
--- a/packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.tsx
+++ b/packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.tsx
@@ -78,13 +78,25 @@ export const StaticDatePicker = React.forwardRef(function StaticDatePicker<
     displayStaticWrapperAs = 'mobile',
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
 
   const { pickerProps, inputProps, wrapperProps } = usePickerState(props, datePickerValueManager);
   const validationError = useDateValidation(props) !== null;
 
-  const DateInputProps = { ...inputProps, ...other, ref, validationError, components };
+  const DateInputProps = {
+    ...inputProps,
+    ...other,
+    label,
+    inputFormat,
+    renderInput,
+    ref,
+    validationError,
+    components,
+  };
 
   return (
     <PickerStaticWrapper
diff --git a/packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx b/packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx
index 43a8b420..5530a0f2 100644
--- a/packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx
+++ b/packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx
@@ -78,6 +78,9 @@ export const StaticDateTimePicker = React.forwardRef(function StaticDateTimePick
     value,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
 
@@ -91,6 +94,9 @@ export const StaticDateTimePicker = React.forwardRef(function StaticDateTimePick
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     ref,
     validationError,
     components,
diff --git a/packages/x-date-pickers/src/StaticTimePicker/StaticTimePicker.tsx b/packages/x-date-pickers/src/StaticTimePicker/StaticTimePicker.tsx
index afd34c5c..d64e365f 100644
--- a/packages/x-date-pickers/src/StaticTimePicker/StaticTimePicker.tsx
+++ b/packages/x-date-pickers/src/StaticTimePicker/StaticTimePicker.tsx
@@ -76,6 +76,9 @@ export const StaticTimePicker = React.forwardRef(function StaticTimePicker<
     value,
     components,
     componentsProps,
+    label,
+    inputFormat,
+    renderInput,
     ...other
   } = props;
 
@@ -85,6 +88,9 @@ export const StaticTimePicker = React.forwardRef(function StaticTimePicker<
   const DateInputProps = {
     ...inputProps,
     ...other,
+    label,
+    inputFormat,
+    renderInput,
     ref,
     validationError,
     components,

```

Reply:
As far as I know `DayPicker` is not exported, so it can be let as it is for now.

Already have some refactoring in #5820 to have similar implementation between `DayPicker`, `MonthPicker`, and `YearPicker`

Reply:
Yeah it's a heavy refactoring. We don't need to do it in this PR but if we want to follow the convention we need to think about it later.